### PR TITLE
use model_dir as default ckpt path

### LINKED ISSUE #612: Unable to do inference using Keras model
Hi,

First of all, thanks for the great work. This model is of great use to me.

I am trying to use the new Keras model to do inference (using the `infer.py` script). However, I get some errors.

## Command

`python keras\infer.py --image_path C:\Users\dmurray\Pictures\video_frame2020-07-20-15h37m23s577.png --output_dir "C:\Users\dmurray\Desktop\kerastest" --checkpoint "C:\Users\dmurray\Desktop\kerastest\efficientdet-d0"`

The folder `kerastest\efficientdet-d0` contains the D0 model checkpoint (extracted today). The folder contains the following files:
`checkpoint  d0_coco_test-dev2017.txt  d0_coco_val.txt  model.data-00000-of-00001  model.index  model.meta`

## Error

```python
(analyses_images) C:\Users\dmurray\code\analyses_images\social_distancing\automl\efficientdet>python keras\infer.py --image_path C:\Users\dmurray\Pictures\video_frame2020-07-20-15h37m23s577.png --output_dir "C:\Users\dmurray\Desktop\kerastest" --checkpoint "C:\Users\dmurray\Desktop\kerastest\efficientdet-d0"
2020-07-20 15:46:25.602061: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_102.dll'; dlerror: cudart64_102.dll not found
2020-07-20 15:46:25.605299: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2020-07-20 15:46:28.939468: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'nvcuda.dll'; dlerror: nvcuda.dll not found
2020-07-20 15:46:28.943000: E tensorflow/stream_executor/cuda/cuda_driver.cc:351] failed call to cuInit: UNKNOWN ERROR (303)
2020-07-20 15:46:28.946909: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: AMT34204
2020-07-20 15:46:28.949207: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: AMT34204
2020-07-20 15:46:34.293075: W tensorflow/core/util/tensor_slice_reader.cc:95] Could not open C:\Users\dmurray\Desktop\kerastest\efficientdet-d0: Unknown: NewRandomAccessFile failed to Create/Open: C:\Users\dmurray\Desktop\kerastest\efficientdet-d0 : Access refused.
; Input/output error
Traceback (most recent call last):
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\tensorflow_core\python\training\py_checkpoint_reader.py", line 95, in NewCheckpointReader
    return CheckpointReader(compat.as_bytes(filepattern))
RuntimeError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "keras\infer.py", line 99, in <module>
    app.run(main)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\absl\app.py", line 299, in run
    _run_main(main, args)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\absl\app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "keras\infer.py", line 69, in main
    model.load_weights(FLAGS.checkpoint)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\tensorflow_core\python\keras\engine\training.py", line 234, in load_weights
    return super(Model, self).load_weights(filepath, by_name, skip_mismatch)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\tensorflow_core\python\keras\engine\network.py", line 1187, in load_weights
    py_checkpoint_reader.NewCheckpointReader(filepath)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\tensorflow_core\python\training\py_checkpoint_reader.py", line 99, in NewCheckpointReader
    error_translator(e)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\tensorflow_core\python\training\py_checkpoint_reader.py", line 48, in error_translator
    raise errors_impl.OpError(None, None, error_message, errors_impl.UNKNOWN)
tensorflow.python.framework.errors_impl.OpError
```

The access refused error is quite odd because my user has readwrite access to the folder.

I also tried pointing directly to the `checkpoint` file inside the model, but now I get the following error:

```python
(analyses_images) C:\Users\dmurray\code\analyses_images\social_distancing\automl\efficientdet>python keras\infer.py --image_path C:\Users\dmurray\Pictures\video_frame2020-07-20-15h37m23s577.png --output_dir "C:\Users\dmurray\Desktop\kerastest" --checkpoint "C:\Users\dmurray\Desktop\kerastest\efficientdet-d0\checkpoint"
2020-07-20 15:56:08.333181: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_102.dll'; dlerror: cudart64_102.dll not found
2020-07-20 15:56:08.336849: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2020-07-20 15:56:11.559967: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'nvcuda.dll'; dlerror: nvcuda.dll not found
2020-07-20 15:56:11.563366: E tensorflow/stream_executor/cuda/cuda_driver.cc:351] failed call to cuInit: UNKNOWN ERROR (303)
2020-07-20 15:56:11.568406: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: AMT34204
2020-07-20 15:56:11.571289: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: AMT34204
2020-07-20 15:56:17.327382: W tensorflow/core/util/tensor_slice_reader.cc:95] Could not open C:\Users\dmurray\Desktop\kerastest\efficientdet-d0\checkpoint: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?
Traceback (most recent call last):
  File "keras\infer.py", line 99, in <module>
    app.run(main)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\absl\app.py", line 299, in run
    _run_main(main, args)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\absl\app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "keras\infer.py", line 69, in main
    model.load_weights(FLAGS.checkpoint)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\tensorflow_core\python\keras\engine\training.py", line 234, in load_weights
    return super(Model, self).load_weights(filepath, by_name, skip_mismatch)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\tensorflow_core\python\keras\engine\network.py", line 1215, in load_weights
    with h5py.File(filepath, 'r') as f:
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\h5py\_hl\files.py", line 408, in __init__
    swmr=swmr)
  File "C:\Users\dmurray\Anaconda3\envs\analyses_images\lib\site-packages\h5py\_hl\files.py", line 173, in make_fid
    fid = h5f.open(name, flags, fapl=fapl)
  File "h5py\_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py\_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py\h5f.pyx", line 88, in h5py.h5f.open
OSError: Unable to open file (file signature not found)
```

## Versions / environnent info
* EfficientDet: commit 6fdd1de778408625c1faf368a327fe36ecd41bf7
* TensorFlow 2.1.0
* Python 3.7.7
* Windows using an Anaconda distribution

Thanks in advance.
David

### PR DESCRIPTION
**Pull request by fsx950223**
Fix #612

### CODE REVIEW

**Comment by mingxingtan** [comment]
Oops, I did not see your PR, and just submitted https://github.com/google/automl/commit/ccf8b6ee6261d5f6e2f6f30c247b82bbcf309210

You can revert the changes for infer.py and keep the changes for eval.py

File: efficientdet/keras/eval.py [hunk 1]
```diff
@@ -26,12 +26,14 @@
...
+import tensorflow as tf
+
 flags.DEFINE_string('val_file_pattern', None,
                     'Glob for eval tfrecords, e.g. coco/val-*.tfrecord.')
 flags.DEFINE_string('val_json_file', None,
                     'Groudtruth file, e.g. annotations/instances_val2017.json.')
 flags.DEFINE_string('model_name', 'efficientdet-d0', 'Model name to use.')
-flags.DEFINE_string('checkpoint', None, 'Location of the checkpoint to run.')
+flags.DEFINE_string('model_dir', None, 'Location of the checkpoint to run.')
 flags.DEFINE_integer('batch_size', 8, 'Batch size.')
```
Comment (mingxingtan) [inline efficientdet/keras/eval.py:37]: Better to allow both dir and filename as shown in https://github.com/google/automl/commit/ccf8b6ee6261d5f6e2f6f30c247b82bbcf309210

**Comment by fsx950223** [comment]
I use model_dir in train.py. It's a little difficult to change to ckpt path.

**Comment by mingxingtan** [comment]
> I use model_dir in train.py. It's a little difficult to change to ckpt path.

Okay, then let's just use model_dir for consistency. You can just override my previous changes.

File: efficientdet/iou_utils.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
 from typing import Union, Text
 import numpy as np
 import tensorflow as tf
+from tensorflow.python.framework.ops import executing_eagerly_outside_functions # pylint:disable=g-direct-tensorflow-import
```
Comment (fsx950223) [inline efficientdet/iou_utils.py:25]: TF1.15 doesn't export executing_eagerly_outside_functions

**Pull request merged**